dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -419,7 +419,7 @@ public Workflow getWorkflow(@ApiParam(hidden = true) @Parameter(hidden = true, n
         @Parameter(name = "include", description = WORKFLOW_INCLUDE_MESSAGE, in = ParameterIn.QUERY) @ApiParam(value = WORKFLOW_INCLUDE_MESSAGE) @QueryParam("include") String include) {
         Workflow workflow = workflowDAO.findById(workflowId);
         checkNotNullEntry(workflow);
-        checkCanRead(user, workflow);
+        checkCanExamine(user, workflow);
```

Comment:
Thinking out loud, this means a published workflow will only be visible to the workflow's owners via the proprietary API. But we have #5061 to track that, if we want to revist.

Reply:
Via that endpoint, yes.

From what I can tell, when the API was designed, the choice was made to create two versions of a lot of the entry-related endpoints: one version for "owners", which retrieves the full entry, and another version for the "public", that returns published entries with the sensitive information removed.   For example, the "published" analog to the above endpoint is:

https://github.com/dockstore/dockstore/blob/7eda964ec43f6648269d437023617cb3b9d1f71d/dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java#L741

One could combine them, and simply censor any info that the user wasn't authorized to see, which would implement the desired behavior you describe in #5061 (I think).   Not a small change, though...

Reply:
FWIW, the proprietary probably came before TRS too. At one point, we tried to migrate much of the UI to TRS and use optional auth there. Depending on how far along that is, #5061 could also attempt to clean up some of these redundant(?) endpoints